lib/util/Components.js

```diff
@@ -859,44 +893,21 @@ function componentRule(rule, context) {
     },
   };
 
-  // Update the provided rule instructions to add the component detection
   const ruleInstructions = rule(context, components, utils);
-  const updatedRuleInstructions = Object.assign({}, ruleInstructions);
   const propTypesInstructions = propTypesUtil(context, components, utils);
   const usedPropTypesInstructions = usedPropTypesUtil(context, components, utils);
   const defaultPropsInstructions = defaultPropsUtil(context, components, utils);
-  const allKeys = new Set(Object.keys(detectionInstructions).concat(
-    Object.keys(propTypesInstructions),
-    Object.keys(usedPropTypesInstructions),
-    Object.keys(defaultPropsInstructions),
-    Object.keys(reactImportInstructions)
-  ));
 
-  allKeys.forEach((instruction) => {
-    updatedRuleInstructions[instruction] = (node) => {
-      if (instruction in detectionInstructions) {
-        detectionInstructions[instruction](node);
```

Comment:
This property access performed for each AST node turned out to be performance bottleneck.